idaes/core/util/model_diagnostics.py

```diff
@@ -17,29 +17,1022 @@
+        description="Tolerance for raising a caution for large Jacobian values.",
+    ),
+)
+CONFIG.declare(
+    "jacobian_large_value_warning",
+    ConfigValue(
+        default=1e8,
+        domain=float,
+        description="Tolerance for raising a warning for large Jacobian values.",
+    ),
+)
+CONFIG.declare(
+    "jacobian_small_value_caution",
+    ConfigValue(
+        default=1e-4,
+        domain=float,
+        description="Tolerance for raising a caution for small Jacobian values.",
+    ),
+)
+CONFIG.declare(
+    "jacobian_small_value_warning",
+    ConfigValue(
+        default=1e-8,
+        domain=float,
+        description="Tolerance for raising a warning for small Jacobian values.",
+    ),
+)
+
+
+@document_kwargs_from_configdict(CONFIG)
+class DiagnosticsToolbox:
+    """
+    The IDAES Model DiagnosticsToolbox.
+
+    To get started:
+
+      1. Create an instance of your model (this does not need to be initialized yet).
+      2. Fix variables until you have 0 degrees of freedom. Many of these tools presume
+         a square model, and a square model should always be the foundation of any more
+         advanced model.
+      3. Create an instance of the DiagnosticsToolbox and provide the model to debug as
+         the model argument.
+      4. Call the ``report_structural_issues()`` method.
+
+    Model diagnostics is an iterative process and you will likely need to run these
+    tools multiple times to resolve all issues. After making a change to your model,
+    you should always start from the beginning again to ensure the change did not
+    introduce any new issues; i.e., always start from the report_structural_issues()
+    method.
+
+    Note that structural checks do not require the model to be initialized, thus users
+    should start with these. Numerical checks require at least a partial solution to the
+    model and should only be run once all structural issues have been resolved.
+
+    Report methods will print a summary containing three parts:
+
+    1. Warnings - these are critical issues that should be resolved before continuing.
+       For each warning, a method will be suggested in the Next Steps section to get
+       additional information.
+    2. Cautions - these are things that could be correct but could also be the source of
+       solver issues. Not all cautions need to be addressed, but users should investigate
+       each one to ensure that the behavior is correct and that they will not be the source
+       of difficulties later. Methods exist to provide more information on all cautions,
+       but these will not appear in the Next Steps section.
+    3. Next Steps - these are recommended methods to call from the DiagnosticsToolbox to
+       get further information on warnings. If no warnings are found, this will suggest
+       the next report method to call.
+
+    Args:
+        model - model to be diagnosed. The DiagnosticsToolbox does not support indexed Blocks.
```

Comment:
<img width="1099" alt="Screenshot 2023-08-24 at 5 27 37 PM" src="https://github.com/IDAES/idaes-pse/assets/8885032/cd8ccf5e-8a70-413d-97bb-eacaec64cbf3">

This is what I see in the docs. Not sure exactly how to get this to format properly. Maybe `model: Model to be diagnosed...`?